### PR TITLE
chore: switch main/03_expressions.lean to semantic linebreaks

### PR DESCRIPTION
**Pull request by jcommelin**

### CODE REVIEW

File: lean/main/03_expressions.lean [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
-which help with the manipulation of expressions. This chapter and the next
-survey the most important ones.
+Expressions (terms of type `Expr`) are an abstract syntax tree for Lean programs.
+This means that each term which can be written in Lean has a corresponding `Expr`.
+For example, the application `f e` is represented by the expression `Expr.app ⟦f⟧ ⟦e⟧`,
+where `⟦f⟧` is a representation of `f` and `⟦e⟧` a representation of `e`.
+Similarly, the term `Nat` is represented by the expression ``Expr.const `Nat []``.
+(The backtick and empty list are discussed below.)
+
+The ultimate purpose of a Lean tactic block
```
Comment (Julian) [inline lean/main/03_expressions.lean:11]: Splitting this line here is intentional?

I personally tend to use semantic linefeeds but essentially nearly *never* break on anything but sentence endings (.!?) even though yeah breaking on a , or ; is allowed/can be fine and essentially meets the "Reason" for using semantic linefeeds.

But breaking on spaces is a bit more iffy I think.

File: lean/main/03_expressions.lean [hunk 2]
```diff
@@ -52,139 +51,155 @@ end Playground
...
+- `fvar` is a __free variable__.
+  These are variables which are not bound by a binder.
+  An example is `x` in `x + 2`.
+  Note that you can't just look at a free variable `x` and tell what its type is,
+  there needs to be a context which contains a declaration for `x` and its type.
+  A free variable has an ID that tells you where to look for it in a `LocalContext`.
+  In Lean 3, free variables were called "local constants" or "locals".
+- `mvar` is a __metavariable__.
+  There will be much more on these later,
+  but you can think of it as a placeholder or a 'hole' in an expression
```
Comment (Julian) [inline lean/main/03_expressions.lean:69]: Same here.

File: lean/main/03_expressions.lean [hunk 3]
```diff
@@ -52,139 +51,155 @@ end Playground
...
-  $t, $b`. Note that the non-dependent arrow `α → β` is a special case of `(a :
-  α) → β` where `β` doesn't depend on `a`. The `E` on the end of `forallE` is to
-  distinguish it from the `forall` keyword.
+- `app` is a function application.
+  Multiple arguments are done using _partial application_: `f x y ↝ app (app f x) y`.
+- `lam n t b` is a lambda expression (`fun ($n : $t) => $b`).
+  The `b` argument is called the __body__.
+  Note that you have to give the type of the variable you are binding.
+- `forallE n t b` is a dependent arrow expression (`($n : $t) → $b`).
+  This is also sometimes called a Π-type or Π-expression
```
Comment (Julian) [inline lean/main/03_expressions.lean:79]: And here.

File: lean/main/03_expressions.lean [hunk 4]
```diff
@@ -52,139 +51,155 @@ end Playground
...
+  and is often written `∀ $n : $t, $b`.
+  Note that the non-dependent arrow `α → β` is a special case of `(a : α) → β`
```
Comment (Julian) [inline lean/main/03_expressions.lean:81]: And here (I'm going to stop now :D)

**Comment by jcommelin** [comment]
@Julian I think I've internalized a version of semantic line-breaks that aims for <100 chars per line. So I tend to linebreak sentences into clauses when they get too long. The benefit is that you can have the info view open on a smaller screen, and everything still fits.
But maybe we just want one line per sentence, and people can word-wrap if their screen is too small...

**Comment by Julian** [comment]
Got it, yeah I tend to go for the latter ("it's prose now, so use word wrap if you need to") but you're right, it's definitely not the only way, so perhaps never mind my comments!

**Comment by Julian** [comment]
In light of the above by the way, I have no personal issues with this change, so consider it a LGTM from me if you want to merge.

**Pull request merged**